runtimes/rust/src/call_context.rs

```diff
@@ -88,16 +114,14 @@ impl<'a> CallContext<'a> {
     ///
     /// This method can only be called by the supervisor; the call will panic otherwise.
     #[doc(hidden)]
+    #[deprecated]
```

Comment:
It isn't deprecated right now, is it? Currently it's the only API for service registration that we have, AFAIK.

Reply:
Same applies to other methods marked as `deprecated`.

Reply:
I use deprecated marker to mark methods which I want to remove by this PR.

Reply:
Got it. With a comment it would have been more clear :)